docs/query.md

```diff
@@ -0,0 +1,314 @@
+Objects can be nested, and other clauses used, to shape the exact format needed in the result.
+
+```javascript
+const result = await get({
+  $id: 'muASxsd3',
+  title: true,
+  director: true,
+  technicalData: {
+    runtime: true,
+    color: true,
+    aspectRatio: true
+  }
+})
+```
+
+### `$id`: _string_
+
+Id of the object to get.  
+If omited, the _root_ object is returned. Can only be used at the root of the query object.
+
+```javascript
+const result = await get({
+  $id: 'muASxsd3'
+})
+```
+
+### `$all`: _boolean_
+
+Includes all the fields for the obect at the level where the clause is used.  
+[References](./glossary.md#references) are not included. (?)
+
+```javascript
+const result = await get({
+  $id: 'peASxsd3',
+  $all: true
+})
+```
+
+Fields can be excluded if a false is set to specific fields.
+```javascript
+const result = await get({
+  $id: 'peASxsd3',
+  $all: true,
+  died: false
+})
+```
+[See test](../client/test/examples/clauses/all.ts)
+
+### `$value`: _any_
+
+Overrides the current value of the field.
+
+```javascript
+const result = await get({
+  $id: 'moASxsd3',
+  title: { $value'Amazing movie' }
+})
+```
+[See test](../client/test/examples/clauses/value.ts)
+
+### `$default`: _any_
+
+Default value to be returned in case the field has no value set.
+
+```javascript
+const result = await get({
+  $id: 'moASxsd3',
+  director: { $default: 'Unknown director' }
+})
+```
+[See test](../client/test/examples/clauses/default.ts)
+
+### `$language`: _string_
+
+Filters document data to a set language on fields that supoprt it.
+
+```javascript
+const result = await client.get({
+  $language: 'en',
+  $id: 'mo2001ASpaceOdyssey',
+  title: true
+})
+```
+[See test](../client/test/examples/clauses/language.ts)
+
+### `$field`: _string_, _Array&lt;string&gt;_
+
+The `$field` clause is used to alias a field to another field in the same document.
+
+```javascript
+const result = await client.get({
+  $id: 'mo2001ASpaceOdyssey',
+  directedBy: { $field: 'director' }
+})
+```
+
+Dot notation can be used to create a path and alias a nested field or even specific data inside a _JSON datatype_.
+```javascript
+const result = await client.get({
+  $id: 'mo2001ASpaceOdyssey',
+  ratio: { $field: 'technicalData.aspectRatio' }
+})
+```
+
+Because _text datatype_ fields interanlly are a simple nested object, the `$field` clause can be used to get a specific language from a text field.
+```javascript
+const result = await client.get({
+  $id: 'mo2001ASpaceOdyssey',
+  englishTitle: { $field: 'title.en' }
+})
+```
+
+The `$field` clause can take an Array instead of a string. The array can have several field names or paths for the alias to point to. The first that is defined in the document will be returned.
+
+```javascript
+const = await client.get({
+  $id: 'mo2001ASpaceOdyssey',
+  by: { $field: ['producer', 'director'] }
+})
+```
+
+[See test](../client/test/examples/clauses/field.ts)
+
+### `$inherit`: _boolean_
```

Comment:
Here we could also explain by type/name inheritance and inherit as item

Reply:
how does that work? have an example?

Reply:
Inherit item: https://github.com/atelier-saulx/selva/blob/master/client/test/getBasic.ts#L1021

Reply:
By type: https://github.com/atelier-saulx/selva/blob/master/client/test/getField.ts#L479

Reply:
By name: https://github.com/atelier-saulx/selva/blob/master/client/test/getField.ts#L530 (since name is also non-unique)